src/pynwb/validate.py

```diff
@@ -26,6 +26,52 @@ def _validate_helper(**kwargs):
     return (errors is not None and len(errors) > 0)
 
 
+def get_cached_namespaces_to_validate(path):
+    """
+    Determine the most specific namespace(s) (i.e., extensions) that are cached in the given
+    NWB file that should be used for validation.
+
+    Example
+    -------
+
+    The following example illustrates how we can use this function to validate against namespaces
+    cached in a file. This is useful, e.g., when a file was created using an extension
+
+    >>> from pynwb import validate
+    >>> from pynwb.validate import get_cached_namespaces_to_validate
+    >>> path = "my_nwb_file.nwb"
+    >>> validate_namespaces, manager, cached_namespaces = get_cached_namespaces_to_validate(path)
+    >>> with NWBHDF5IO(path, "r", manager=manager) as reader:
+    >>>     errors = []
+    >>>     for ns in validate_namespaces:
+    >>>         errors += validate(io=reader, namespace=ns)
+
+    :param path: Path for the NWB file
+    :return: Tuple with:
+      - List of strings with the most specific namespace(s) to use for validation.
+      - BuildManager object for opening the file for validation
+      - Dict with the full result from NWBHDF5IO.load_namespaces
+    """
+    catalog = NamespaceCatalog(NWBGroupSpec, NWBDatasetSpec, NWBNamespace)
+    ns_deps = NWBHDF5IO.load_namespaces(catalog, path)
+    # determine which namespaces are the most specific (i.e. extensions) and validate against those
+    s = set(ns_deps.keys())
+    for k in ns_deps:
+        s -= ns_deps[k].keys()
+    # TODO remove this workaround for issue https://github.com/NeurodataWithoutBorders/pynwb/issues/1357
+    if 'hdmf-experimental' in s:
+        s.remove('hdmf-experimental')  # remove validation of hdmf-experimental for now
+    namespaces = list(sorted(s))
```

Comment:
```suggestion
    namespaces = sorted(s)
```

`sorted()` already returns a list.